openstack_controller/datadog_checks/openstack_controller/components/compute.py

```diff
@@ -233,67 +233,80 @@ def _report_quota_sets(self, project_id, tags, config):
+                    lambda: get_compute_servers_func,
+                    limit=config_servers.get('limit'),
+                    include=config_servers_include,
+                    exclude=config_servers.get('exclude'),
+                    interval=config_servers.get('interval'),
+                    key=lambda server: server.get('name'),
+                )
+        if servers_discovery:
+            discovered_servers = list(servers_discovery.get_items())
+        else:
+            discovered_servers = [(None, server.get('name'), server, None) for server in get_compute_servers_func]
+        aggregates = self.check.api.get_compute_aggregates()
+        self.check.log.debug("aggregates: %s", aggregates)
+        for _pattern, _item_name, item, item_config in discovered_servers:
+            self.check.log.debug("item: %s", item)
+            self.check.log.debug("item_config: %s", item_config)
+            server = get_metrics_and_tags(
+                item,
+                tags=NOVA_SERVER_TAGS,
+                prefix=NOVA_SERVER_METRICS_PREFIX,
+                metrics=NOVA_SERVER_METRICS,
+                lambda_name=lambda key, item=item: (
+                    'active'
+                    if key == 'status' and item['status'] == 'ACTIVE'
+                    else 'error' if key == 'status' and item['status'] == 'ERROR' else key
+                ),
+                lambda_value=lambda key, value, item=item: (
+                    1 if key == 'status' and (item['status'] == 'ACTIVE' or item['status'] == 'ERROR') else value
+                ),
+            )
+            self.check.log.debug("server: %s", server)
+            all_tags = list(set(tags + server['tags']))
+            self.check.gauge(NOVA_SERVER_COUNT, 1, tags=all_tags, hostname=item['id'])
+            for metric, value in server['metrics'].items():
+                self.check.gauge(metric, value, tags=all_tags, hostname=item['id'])
+            collect_flavors = item_config.get('flavors', True) if item_config else True
+            if collect_flavors:
+                self._report_server_flavor(item, all_tags)
+            collect_diagnostics = item_config.get('diagnostics', True) if item_config else True
+            if collect_diagnostics:
+                self._report_server_diagnostics(item, all_tags)
+            report_external_tags = item_config.get('external_tags', True) if item_config else True
+            if report_external_tags:
+                self._report_external_tags(item, aggregates)
+
+    @Component.register_global_metrics(ID)
+    @Component.http_error()
+    def _report_all_project_servers(self, config, tags):
+        report_servers = True
+        config_servers = config.get('servers', {})
+        if isinstance(config_servers, bool):
+            report_servers = config_servers
+            config_servers = {}
+        if report_servers and config_servers.get('all_projects', config.get('all_projects', False)):
```

Comment:
I'm not sure why there are two different places for setting the `all_projects` option in the config (one under the "server" component and one under instance options). 

Would it be possible to just provide one location to specify the `all_projects` option, to simplify the logic logic like `config_servers.get('all_projects', config.get('all_projects', False))` to something like`config.get('all_projects', False)`? 

If 2 different locations for the option is absolutely required, then perhaps could the naming of `config_servers/all_projects` and `config/all_projects` be more differentiated? It is a bit difficult to understand this logic, as someone who is just now reading this code for the first time.

Reply:
Good point!
I've simplified the config.
Limited the `all_projects` config key just to `instance` level